proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,197 @@
+  string page_token = 4;
+}
+
+// Response to list members.
+message ListMembersResponse {
+  // ID of the group owner.
+  string owner_account_id = 1;
+
+  // List of members of the group.
+  map<string, Role> members = 2;
+
+  // Token to continue listing members from.
+  string next_page_token = 3;
+}
+
+// Request to list content.
+message ListContentRequest {
+  // Required. ID of the group to list content of.
+  string id = 1;
+
+  // Optional. Version of the group to list content of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of content to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing content from.
+  string page_token = 4;
+}
+
+// Response to list content.
+message ListContentResponse {
+  // List of content of the group.
+  // Key is the path and value is a Hyperdocs URL to the content.
+  map<string, string> content = 1;
+
+  // Token to continue listing content from.
+  string next_page_token = 2;
+}
+
+// Request to list groups.
+message ListGroupsRequest {
+  // Optional. Maximum number of groups to return.
+  int32 page_size = 1;
+
+  // Optional. Page token to continue listing groups from.
+  string page_token = 2;
+}
+
+// Response to list groups.
+message ListGroupsResponse {
+  // List of groups.
+  repeated Group groups = 1;
+
+  // Token to continue listing groups from.
+  string next_page_token = 2;
+}
+
+// Group represents the basic information about a group entity.
+message Group {
+  // ID of the group.
+  string id = 1;
+
+  // Title of the group.
+  string title = 3;
+
+  // Description of the group.
+  string description = 4;
+
+  // Account ID of the group owner.
+  string owner_account_id = 5;
+
+  // Timestamp when the group was first created.
+  google.protobuf.Timestamp create_time = 6;
+
+  // Latest version of the group according to the information from its owner and members
+  // that we happen to have locally. Most of the time this is the version that should be used
+  // to get the most recent group information.
+  string latest_version = 7;
+
+  // Latest version of the group according to our trusted contacts.
+  // It's possible that none of the group members is our trusted contact,
+  // but still some of our trusted contacts might mutate the group entity for themselves,
+  // in which case care must be takes while displaying the group entity to the user.
+  // If none of the group members is our trusted contact, this version will be empty.
+  // If only a subset of group members are our trusted contacts, this will be the head changes produced by them.
+  string latest_local_trusted_version = 8;
+
+  // Latest version of the group entity according to all the peers we happen to know about.
+  // If no peer outside the group members produced any changes to the entity, this should be the same as latest_version.
+  string latest_local_public_version = 9;
```

Comment:
If it's in the API, doesn't mean it shows in the UI :) As with `latest_local_trusted_version`, this is here just because we can. Doesn't hurt to have it, even though we'll probably never need it, because our software wouldn't let to modify the group entity for non-members.

The only thing that might occur is that en editor concurrently might create a change with owner removing that editor. In this case we might have a dangling head that would appear here, but not in the `latest_version`.

Again, for our current purposes we should only be using `latest_version`.

Reply:
Can we remove this field? Extra unused fields are just super confusing IMHO. It takes all this effort to understand and maintain. We wouldn't even be having this conversation. There is a lot of extra data that you have in the go world- lets avoid exposing that on the API unless the UI needs it

Reply:
I disagree (just a little bit 🙂). I answered here in the other discussion https://github.com/mintterteam/mintter/pull/1408#discussion_r1284735674. Closing this one, let's talk about it over there.